src/resolver.ts

```diff
@@ -1275,6 +1275,8 @@ export class Resolver extends DiagnosticEmitter {
       case ElementKind.FIELD: { // someVar.prop
         let variableLikeElement = <VariableLikeElement>target;
         let type = variableLikeElement.type;
+        let narrowedType = ctxFlow.getNarrowedType(variableLikeElement);
+        if (narrowedType) type = narrowedType;
```

Comment:
Applying type narrowing on fields is risky because any intermediate function call (or property access with a getter/setter) can have the side-effect of assigning a different value to the field, so the narrowed type would no longer hold and the program would statically produce failures. So far, we limited such checks to locals, since in the absence of closures, these cannot be mutated as a side-effect. As such, this change is likely unsafe.

Reply:
Yes, perhaps we should comment out with some TODO for now, until we have the necessary infra to check the side effects for Flow. Calling a method to instance this reference or passing the reference to a global function between entry and exit points of this path will be considered a side-effect that will either stop type narrowing or start some kind of approximation or deeper analysis.

Reply:
Ok, I will remove the logic of global and field. And then mark it as TODO.